subsys/fs/Kconfig

```
@@ -67,6 +67,64 @@ config FS_FATFS_NUM_FILES
 config FS_FATFS_NUM_DIRS
 	int "Maximum number of opened directories"
 	default 4
+
+config FS_FATFS_LFN
+	bool "Enable long filenames (LFN)"
+	default n
+	help
+	  Without long filenames enabled, file names are limited to 8.3 format.
+	  This option increases working buffer size.
+
+config FS_FATFS_LFN_MODE
+	int "Long filename memory mode"
+	depends on FS_FATFS_LFN
+	range 1 3
+	default 1
+	help
+	  Valid memory mode values:
+	  1 - Enable LFN with static working buffer on the BSS.
+	  2 - Enable LFN with dynamic working buffer on the STACK.
```

Where is option 2 and 3 used, in the FATFS repo I only see use of option 1?

---

> 
> 
> Where is option 2 and 3 used, in the FATFS repo I only see use of option 1?

The help text is taken from ffconf.h:
https://github.com/zephyrproject-rtos/fatfs/blob/0c1288bb817c1976388e0f55e8483f671ae437ab/include/ffconf.h#L101

And the macros that determine location of the buffer are in ff.c:
https://github.com/zephyrproject-rtos/fatfs/blob/0c1288bb817c1976388e0f55e8483f671ae437ab/ff.c#L554

---

The FATFS code only assigns a value to `_USE_LFN` if `FS_FATFS_LFN` is set to 1.

```
#if defined(CONFIG_FS_FATFS_LFN) && CONFIG_FS_FATFS_LFN == 1
#define	_USE_LFN	CONFIG_FS_FATFS_LFN_MODE
#else
```

---

Using the suggested kconfig choice for the "memory mode" option will make this more clear, I think. Instead of using a FS_FATFS_LFN_MODE integer assigned directly to _USE_LFN, I plan to update the fatfs pull request to use a more concise value for _USE_LFN:
(CONFIG_FS_FATFS_LFN_MODE_x are choice options here)

```suggestion
#if CONFIG_FS_FATFS_LFN_MODE_BSS
#define	_USE_LFN	1
#elif CONFIG_FS_FATFS_LFN_MODE_STACK
#define	_USE_LFN	2
#elif CONFIG_FS_FATFS_LFN_MODE_HEAP
#define	_USE_LFN	3
#else
#error Invalid LFN buffer location
#endif /* CONFIG_FS_FATFS_LFN_MODE_BSS */
```